docs/DeveloperGuide.md

```diff
@@ -1,34 +1,972 @@
+
+#### 3.4.1 `module` Package
+
+##### Description
+
+The `module` package consists of classes that are used to define the type of data to be stored in a module object and establish 
+a framework to show how other components can make use of the features in module classes.
+
+##### Design
+
+The `module` package consists of classes related to module objects. An abstract class `Module` is created to hold attributes 
+and methods applicable to all class objects. It is then inherited by all other child module classes. A class diagram illustrating 
+the relationship between the interaction of classes under the module package is shown below.
+
+![archi](images/ModuleClassDiagram.png)
+
+<sup>***Figure 3.4.1.1** UML class diagram for Module package*</sup>
+
+The following child classes are created to handle different types of modules based on the generic module type available in 
+the university:
+- `CoreModule`
+- `GeModule`
+- `ElectiveModule`
+- `MathModule`
+
+Each of the module classes consists of:
+- Attributes related to the module type it is representing
+- Getter and setter methods for setting and retrieval of its attributes
+- Methods that alter an instance of its own class
+
+#### 3.4.1.1 `Module` Class
+
+`Module` class is an abstract class in the module package. It holds the attributes and methods for manipulating the attributes 
+applicable to all modules. The attributes found in the `Module` class are:
+
+Scope   | Type              | Variable             | Description | 
+--------|-------------------|----------------------|-------------|
+private | String            | code                 | Module code of the module object.
+private | String            | name                 | Module name of the module object.
+private | double            | credit               | Modular credit of the module object.
+private | String            | status               | Status of the module, whether it is “taken”, “not taken” or “taking”.
+private | String            | grade                | The grade of taken modules.
+private | ArrayList<String> | preRequisites        | A list of prerequisite modules.
+private | ArrayList<String> | untakenPreRequisites | A list of unsatisfied prerequisite modules.
+private | ArrayList<String> | requiredByModules    | A list of modules requiring the current module as a prerequisite.
+
+The `Module` class also consists of methods that set and get the value of attributes shown in the table above. There are 
+four additional methods in the class, namely `removeUntakenPreRequisite`, `removeRequiredByModule`, `getStatusIcon`and `toString`
+. The `removeUntakenPreRequisite` and `removeRequiredByModule` methods are used to remove a single`untakenPreRequisites` module 
+and `requiredByModules` module respectively, whereas `getStatusIcon` returns the status icon based on the module 
+status. For customized formatting of module printing messages, `toString` method is overridden.
+
+#### 3.4.1.2 `CoreModule` Class
```

Comment:
Should you avoid using 'toString' in all of the subclass explanation? Since you already mentioned it in the parent class explanation, is it enough to just stated that all the subclass contains the 'toString' method?